### PR TITLE
AF-1739: C3 Area Chart is not displayed

### PR DESCRIPTION
**Pull request by jesuino**
(adding also stacked area chart support)

### CODE REVIEW

**Comment by jesuino** [comment]
@ederign @jhrcek would you please review?

Unfortunately I did not find better ways to fix this than looking up the element itself and fixing the opacity. I could also generate the chart only when the parent div is added, but this could create a mess only for fixing the issue.

Thanks!

**Comment by jesuino** [comment]
@ederign @jhrcek had to update the commit! Found an issue, the area was going away again when the page was resized and I found a C3 callback which I use to correct the area everytime the chart is displayed.

**Comment by jesuino** [comment]
Please do not merge this, I found a better solution for this issue, working on it!

**Comment by jesuino** [comment]
the other solution was not solving all scenarios, leaving the original solution.

File: dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3Renderer.java [hunk 1]
```diff
@@ -163,8 +151,12 @@ private C3Displayer getPieChartForSubType(DisplayerSubType subtype) {
     }
     
     private C3Displayer getAreaChartForSubType(DisplayerSubType subtype) {
-        return beanManager.lookupBean(C3AreaChartDisplayer.class)
+        C3AreaChartDisplayer areaChartDisplayer = beanManager.lookupBean(C3AreaChartDisplayer.class)
                 .newInstance();
+        if (subtype == AREA_STACKED) {
```
Comment (ederign) [inline dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3Renderer.java]: Tests.

File: dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3XYDisplayer.java [hunk 2]
```diff
@@ -49,5 +52,34 @@ private void applyPropertiesToAxes(C3AxisInfo axis) {
             axis.getY().setLabel(yLabel);
         }
     }
+    
+    protected String[][] stackedGroups() {
+        String[][] groups;
```
Comment (ederign) [inline dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3XYDisplayer.java]: Tests.

File: dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/area/C3AreaChartView.java [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.NodeList;
 
 public class C3AreaChartView 
        extends C3DisplayerView<C3AreaChartDisplayer> 
        implements C3AreaChartDisplayer.View{
 
+    @Override
+    public void updateChart(C3ChartConf conf) {
+        conf.setOnrendered(() -> fixAreaOpacity());
```
Comment (ederign) [inline dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/area/C3AreaChartView.java]: Tests.

File: dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/area/C3AreaChartView.java [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+
     @Override
     public String getType() {
         return "area";
     }
+    
+    private void fixAreaOpacity() {
+        // This is a workaround for: https://github.com/c3js/c3/issues/2551
+        if (chart != null) {
+            NodeList<Element> paths = chart.getElement().getElementsByTagName("path");
```
Comment (ederign) [inline dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/area/C3AreaChartView.java]: Tests.

**Comment by ederign** [comment]
please jenkins retest this

**Comment by jesuino** [comment]
please jenkins retest this

**Pull request merged**